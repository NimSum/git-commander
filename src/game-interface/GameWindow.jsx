import React, { Component } from 'react'
import { ProgressIndicator } from './ProgressIndicator';

export class GameWindow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shipPosition: 1,
      clearPath: Math.round(Math.random() * (5 - 1) + 1),
      obstaclePositions: [1, 2, 3, 4, 5]
    }
  }

  generateObstacles() {
    const obstacleContainer = []
    for (let i = 0; i <= this.props.currRound; i++) {
      obstacleContainer.push(
        ( <img 
          className={`obstacle obstacle-pos-${i+1}${Math.round(Math.random())}`}
          src="http://clipart-library.com/image_gallery/n756983.gif" 
          alt="obstacle" />)
      )
    }
    return obstacleContainer;
  }

  render() {
    return (
      <section className="game-window">
        {this.generateObstacles()}
        <img 
          className={`octo-ship ship-position-${this.state.shipPosition}` }
          src={ require("../images/jetpacktocat.png") }
          alt="obstacle" />
        < ProgressIndicator 
          currRound={ this.props.currRound } />
      </section>
    )
  }
}

export default GameWindow
