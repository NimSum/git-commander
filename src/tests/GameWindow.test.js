import React from 'react';
import GameWindow from '../game-interface/GameWindow';
import Adapter from 'enzyme-adapter-react-16';
import { shallow , configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('Game Window' , () => {
  

  let wrapper;
  const currRound = 1;

  beforeEach(() => {
    wrapper = shallow(
      < GameWindow 
        currRound={ currRound }
      />
    );
  });

  it.skip('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should have untouched default state when mounted', () => {
    expect(wrapper.state().clearPath <= 5).toEqual(true);
    expect(wrapper.state().moveObstacle).toEqual(false);
  })

  it('Should generate obstacles and track their position', () => {
    expect(wrapper.state().obstacles).toHaveLength(4)
    expect(wrapper.state().obstaclePositions.length).toBeGreaterThan(2);
  })

  it('Should set octocat to be on a collision course with the obstacles', () => {
    wrapper.state().obstaclePositions = ['31'];
    wrapper.instance().setCollitionCourse();
    expect(wrapper.state().octoCatPosition).toEqual(3)
  })

  it('Should be able to move octocat', () => {
    wrapper.state().clearPath = 4;
    wrapper.state().octoCatPosition = 2;
    wrapper.instance().moveOctocat();
    expect(wrapper.state().octoCatPosition).toEqual(4);
  })

})