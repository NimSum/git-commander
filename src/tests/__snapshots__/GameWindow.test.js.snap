// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Window Should match snapshot 1`] = `
<section
  className="game-window"
>
  <div
    className="obstacle-container undefined"
  >
    <img
      alt="obstacle"
      className="obstacle obstacle-pos-31"
      key="0"
      src="http://clipart-library.com/image_gallery/n756983.gif"
    />
    <img
      alt="obstacle"
      className="obstacle obstacle-pos-20"
      key="1"
      src="http://clipart-library.com/image_gallery/n756983.gif"
    />
    <img
      alt="obstacle"
      className="obstacle obstacle-pos-50"
      key="2"
      src="http://clipart-library.com/image_gallery/n756983.gif"
    />
    <img
      alt="obstacle"
      className="obstacle obstacle-pos-21"
      key="3"
      src="http://clipart-library.com/image_gallery/n756983.gif"
    />
  </div>
  <img
    alt="obstacle"
    className="octo-ship ship-position-5"
    src="jetpacktocat.png"
  />
  <ProgressIndicator
    currRound={1}
  />
</section>
`;
